src/app/shared/components/template/components/task-progress-bar/task-progress-bar.component.ts

```diff
@@ -88,10 +182,28 @@ export class TmplTaskProgressBarComponent extends TemplateBaseComponent implemen
     if (previousHighlightedTaskGroup !== newHighlightedTaskGroup) {
       this.campaignService.scheduleCampaignNotifications();
     }
+    this.cdr.markForCheck();
+  }
+
+  async setTaskGroupCompletedStatus(isCompleted: boolean) {
+    console.log(`Setting ${this.params.taskGroupCompletedField} to ${isCompleted}`);
+    await this.templateFieldService.setField(this.params.taskGroupCompletedField, `${isCompleted}`);
   }
 
-  async setTaskGroupCompletedStatus(taskGroupCompletedField: string, isCompleted: boolean) {
-    console.log(`Setting ${taskGroupCompletedField} to ${isCompleted}`);
-    await this.templateFieldService.setField(taskGroupCompletedField, `${isCompleted}`);
+  // Adapted from data-items component
+  private async subscribeToData() {
```

Comment:
@chrismclarke Should we also be unsubscribing inside an `ngOnDestroy()` for this component (and for the `data-items` component)? If not, why not?

Reply:
Subscriptions should always be unsubscribed. I struggle to think of any case where this is not preferable (maybe some query likely to be repeatedly reused across the app, but even then probably better to just recreate). So if not present in data-items component would be good to add